order-book/src/main/java/com/market/banica/order/book/util/InterestsPersistence.java

```diff
@@ -0,0 +1,73 @@
+package com.market.banica.order.book.util;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.market.banica.common.util.ApplicationDirectoryUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class InterestsPersistence {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(InterestsPersistence.class);
+
+    private final String interestsFileName;
+    private final Map<String, Set<String>> interestsMap;
+
+    private final Kryo kryoHandle = new Kryo();
+
+    public InterestsPersistence(String interestsFileName, Map<String, Set<String>> interestsMap) {
+        initKryo();
+        this.interestsFileName = interestsFileName;
+        this.interestsMap = interestsMap;
+    }
+
+    public void persistInterests() throws IOException {
```

Comment:
there should be some synchronization here, what will happen if 2 threads write this file at the same time

Reply:
Since we were already using Executors.newSingleThreadExecutor() for starting subscriptions, I decided to avoid the concurrency problem for persistence using it. Currently in OrderBookService, persistence is called only in subscriptionExecutor.execute() which executes in a dedicated single thread. If you believe that making it fully concurrent is better, I'll update it.

Reply:
ok, approved then